common/src/main/java/org/conscrypt/SSLParametersImpl.java

```diff
@@ -629,14 +629,18 @@ private void enablePSKKeyManagerIfRequested(long sslNativePointer, PSKCallbacks
      * sockets. For more information on SNI, see RFC 6066 section 3.
      */
     private boolean isSniEnabledByDefault() {
-        String enableSNI = System.getProperty("jsse.enableSNIExtension", "true");
-        if ("true".equalsIgnoreCase(enableSNI)) {
-            return true;
-        } else if ("false".equalsIgnoreCase(enableSNI)) {
+        try {
+            String enableSNI = System.getProperty("jsse.enableSNIExtension", "true");
+            if ("true".equalsIgnoreCase(enableSNI)) {
+                return true;
+            } else if ("false".equalsIgnoreCase(enableSNI)) {
+                return false;
+            } else {
+                throw new RuntimeException(
+                        "Can only set \"jsse.enableSNIExtension\" to \"true\" or \"false\"");
+            }
+        } catch (SecurityException e) {
             return false;
```

Comment:
The intention was to bypass the SNI stuff on Java 7 App Engine, but since #170 can already handle that case, I've changed it to true.